### ISSUE TITLE
flex doesn't recognize layout-gt-* and uses wrong css styles

### ISSUE THREAD

**Issue by epelc** [angular/material#2695]
If you have the following html and your window is big enough to apply the `layout-gt-sm` css.
```
//Note this out div is only required to see the bug. But "some text" still has incorrect styles if you don't use this outer div with layout="row"
<div layout="row">
    <div layout="column" layout-gt-sm="row" layout-align="space-between center" flex>
        <div layout="row" layout-align="start center" flex="20">
	        //Everything within this div will incorrectly apply the flex styles for layout="column" even when you are above the sm breakpoint
            <span>Some text</span>
        </div>
    </div>
</div>
```

Then the inner row with the `flex` will use the flex css for being within a column instead of on a row. This ends up applying `max-height: 20%;` instead of `flex: 0 0 20%;`. The issue is that `flex` isn't aware of the `layout-gt-sm` so it doesn't apply the right styles.

Here is a working demo http://plnkr.co/edit/oC2wpxu5c4O3DgYjyvYo?p=preview
You must use ie11 or firefox to see this. As in chrome it will show correctly somehow(possibly a chrome bug) but if you inspect it you will see it uses the `max-height` attribute instead of the `flex`.

Notice the red border around `some text` if you inspect it (on a screen bigger then the sm breakpoint) you will see it has `max-height` instead of the `flex` css.

**Comment by ThomasBurleson** [comment]
To expedite resolution, please always test with the latest versions:
*  Angular Material HEAD
  *  Bower Install - https://github.com/angular/bower-material/blob/master/angular-material.js
  *  Bower Install - https://github.com/angular/bower-material/blob/master/angular-material.css 
  *  RAWGIT - https://rawgit.com/angular/bower-material/master/angular-material.js
  *  RAWGIT - https://rawgit.com/angular/bower-material/master/angular-material.css
* AngularJS  Current
  *  Angular 1.3.19 (or greater)
  *  Angular 1.4.6 (or greater).

> Can you confirm the issue still exists with the above versions ? I will reopen if needed.

**Comment by epelc** [comment]
@ThomasBurleson this one is fixed in 0.11.0+
